package-build.el

```diff
@@ -191,16 +182,51 @@ Otherwise do nothing.  FORMAT-STRING and ARGS are as per that function."
     (apply #'message format-string args)))
 
 ;;; Version Handling
+;;;; Common
+
+(defun package-build--select-version (rcp)
+  (pcase-let* ((default-directory (package-recipe--working-tree rcp))
+               (`(,commit ,time ,version)
+                (funcall package-build-get-version-function rcp)))
+    (unless version
+      (error "Cannot detect version for %s" (oref rcp name)))
+    (oset rcp commit commit)
+    (oset rcp time time)
+    (oset rcp version version)))
+
+(cl-defmethod package-build--select-commit ((rcp package-git-recipe) rev exact)
+  (pcase-let*
+      ((`(,hash ,time)
+        (split-string
+         (car (apply #'process-lines
+                     "git" "log" "-n1" "--first-parent"
+                     "--pretty=format:%H %cd" "--date=unix" rev
+                     (and (not exact)
+                          (cons "--" (package-build--spec-globs rcp)))))
+         " ")))
+    (list hash (string-to-number time))))
+
+(cl-defmethod package-build--select-commit ((rcp package-hg-recipe) rev exact)
+  (pcase-let*
+      ((`(,hash ,time ,_timezone)
```

Comment:
I've added this comment:

```
                     ;; The "date" keyword uses UTC. The "hgdate" filter
                     ;; returns two integers separated by a space; the
                     ;; unix timestamp and the time zoneoffset.  We use
                     ;; "hgdate" because that makes it easier to discard
                     ;; the time zone offset, which doesn't interest us.
```

The timezone offset is for the timezone in which to commit was created. I am not adding this information to the comment because we don't care about the timezone.